src/pyprojroot/root.py

```diff
@@ -7,12 +7,15 @@
 
 import pathlib as _pathlib
 import typing as _typing
-from os import PathLike as _PathLike
 
-from .criterion import as_root_criterion as _as_root_criterion
+from .criterion import (
+    as_root_criterion as _as_root_criterion,
+    _CriterionType,
+    _PathType,
+)
 
 
-def as_start_path(start: _PathLike) -> _pathlib.Path:
+def as_start_path(start: _typing.Union[None, _PathType]) -> _pathlib.Path:
```

Comment:
I think it's better to write `_typing.Optional[_PathType]` rather than `_typing.Union[None, _PathType]`. Although in later python versions, `_PathType | None` would be preferred.